### PR TITLE
[zaq1] Add new extractor

### PR DESCRIPTION
**Pull request by piotr5151**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

New extractor for zaq1.pl

### CODE REVIEW

File: youtube_dl/extractor/zaq1.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+        webpage = self._download_webpage(url, video_id)
+
+        title = self._html_search_regex(
+            r'(?s)<h1>\s*<span(?:.+?)class="watch-title"(?:.+?)title="([^"]+)">\1\s*</span>\s*</h1>', webpage, 'title')
+
+        div = re.search(r'(?s)<div(?:.+?)id=(["\'])video_player\1(?:.+?)</div>', webpage).group(0)
+        video_url = self._search_regex(r'data-video-url="(http[^"]+)"', div, 'video url')
+        ext = self._search_regex(r'data-file-extension="([^"]+)"', div, 'ext')
+        duration = self._search_regex(r'data-duration="([^"]+)"', div, 'duration')
```
Comment (dstftw) [inline youtube_dl/extractor/zaq1.py]: None of the optional metadata should break the extraction when missing.

File: youtube_dl/extractor/zaq1.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            'title': title,
+            'formats': [{
+                'url': video_url,
+                'ext': ext,
+                'http_headers': {'Referer': url},
+            }],
+            'thumbnail': thumbnail,
+            'uploader': uploader,
+            'upload_date': upload_date,
+            'duration': duration,
```
Comment (dstftw) [inline youtube_dl/extractor/zaq1.py]: Must be int.

**Comment by piotr5151** [comment]
I hope it's fixed as you suggested.

File: youtube_dl/extractor/zaq1.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        webpage = self._download_webpage(url, video_id)
+
+        title = self._html_search_regex(
+            r'(?s)<h1>\s*<span(?:.+?)class="watch-title"(?:.+?)title="([^"]+)">\1\s*</span>\s*</h1>', webpage, 'title')
```
Comment (dstftw) [inline youtube_dl/extractor/zaq1.py]: Parenthesis are superfluous.

File: youtube_dl/extractor/zaq1.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+        div = self._search_regex(r'(?s)(?P<div><div(?:.+?)id=(["\'])video_player\2(?:.+?)</div>)', webpage, 'video url', group='div')
```
Comment (dstftw) [inline youtube_dl/extractor/zaq1.py]: Parenthesis are superfluous.

File: youtube_dl/extractor/zaq1.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        video_url = self._search_regex(r'data-video-url="(http[^"]+)"', div, 'video url1')
+
+        ext = self._search_regex(r'data-file-extension="([^"]+)"', div, 'ext', None, False)
+        duration = int_or_none(self._search_regex(r'data-duration="([^"]+)"', div, 'duration', None, False))
+        thumbnail = self._search_regex(r'data-photo-url="([^"]+)"', div, 'thumbnail', None, False)
+
+        upload_date = unified_strdate(self._search_regex(r'<div\s+id="watch-uploader-info"><strong\s+class="watch-time-text">Opublikowany\s+([0-9]{4}-[0-9]{2}-[0-9]{2})(?:.+?)</strong></div>', webpage, 'upload date'))
+        uploader = self._search_regex(r'(?s)<div\s+id="watch7-user-header">(?:.+?)Wideo dodał:\s*<a(?:.+?)>\s*([^<]+)\s*</a>', webpage, 'uploader')
```
Comment (dstftw) [inline youtube_dl/extractor/zaq1.py]: All these regexes should be relaxed.

**Comment by piotr5151** [comment]
Changes implemented

**Pull request closed without merging**